mobile/src/components/MosipAuth/index.js

```diff
@@ -0,0 +1,90 @@
+import { BaseFormCard } from "components/BaseFormCard"
+import { CustomButton } from "components/CustomButton"
+import { FORM_WALK_IN_MOSIP_ENROLL_FORM, FORM_WALK_IN_VERIFY_OTP, useWalkInEnrollment } from "components/WalkEnrollments/context";
+import { ApiServices } from "Services/ApiServices";
+import {getMessageComponent, LANGUAGE_KEYS} from "../../lang/LocaleContext";
+
+
+const { useState, useEffect } = require("react")
+
+export const MosipAuth = ({state}) => {
+
+    const [individualId, setIndividualId] = useState("")
+    const [individualIdType, setIndividualIdType] = useState("VID");
+    const [isOTPGenerated, setIsOTPGenerated] = useState(false);
+    const [error, setError] = useState(null);
+    const {goNext} = useWalkInEnrollment();
+    const [maskedEmail, setMaskedEmail] = useState('');
+    const [maskedMobile, setMaskedMobile] = useState('')
+
+    useEffect(() => {
+        if(isOTPGenerated) {
+            goNext(FORM_WALK_IN_MOSIP_ENROLL_FORM, FORM_WALK_IN_VERIFY_OTP, {individualId, individualIdType, from: 'mosip', phone: maskedMobile, maskedEmail});
+        }
+    });
+
+    const handleChange = (e) => {
+        setIndividualIdType(e.target.value);
+    }
+
+    const onGenerateOTP = () => {
+        var regExp = /[a-zA-Z]/g;
```

Comment:
characters are not allowed. we are checking if there is any character present
regarding length, we are not sure how long it can be. so haven't checked that. besides, if the VID/UIN is wrong, MOSIP api will return that it is invalid UID/VIN